pkg/fission-cli/cmd/spec/spec.go

```diff
@@ -197,6 +197,73 @@ func SpecSave(resource interface{}, specFile string) error {
 	return nil
 }
 
+// save saves object encoded value to spec file under given spec directory
+func saveUpdate(data []byte, specDir string, specFile string) error {
+	// verify
+	if _, err := os.Stat(filepath.Join(specDir, "fission-deployment-config.yaml")); os.IsNotExist(err) {
+		return errors.Wrap(err, "Couldn't find specs, run `fission spec init` first")
+	}
+
+	filename := filepath.Join(specDir, specFile)
+	// check if the file is new
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		return errors.Errorf("spec file does not exists")
+	}
+
+	// open spec file to truncate and write
+	f, err := os.OpenFile(filename, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0600)
+	if err != nil {
+		return errors.Wrap(err, "couldn't create spec file")
+	}
+	defer f.Close()
+
+	err = f.Truncate(0)
+	if err != nil {
+		return errors.Wrap(err, "couldn't truncate the spec file")
+	}
+
+	// write our resource
+	_, err = f.Write(data)
+	if err != nil {
+		return errors.Wrap(err, "couldn't write to spec file")
+	}
+	return nil
+}
+
+// called from `fission * update --spec`
+func SpecSaveUpdate(resource interface{}, specFile string) error {
```

Comment:
Can we please write unit test for this function ?

Reply:
Also I would suggest merging this function with the existing `SpecSave` function and capturing whether to consider update or not via flag.